compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangFiniteTypeNode.java

```diff
@@ -81,12 +81,19 @@ public NodeKind getKind() {
     public String toString() {
         StringJoiner stringJoiner = new StringJoiner(" | ");
         for (BLangExpression memberTypeNode : valueSpace) {
-            if (memberTypeNode.getBType().tag == TypeTags.FLOAT) {
-                stringJoiner.add(memberTypeNode.toString() + NumericLiteralSupport.FLOAT_DISCRIMINATOR);
-            } else if (memberTypeNode.getBType().tag == TypeTags.DECIMAL) {
-                stringJoiner.add(memberTypeNode.toString() + NumericLiteralSupport.DECIMAL_DISCRIMINATOR);
-            } else {
-                stringJoiner.add(memberTypeNode.toString());
+            switch (memberTypeNode.getBType().tag) {
+                case TypeTags.FLOAT:
+                    stringJoiner.add(memberTypeNode + NumericLiteralSupport.FLOAT_DISCRIMINATOR);
```

Comment:
Maybe we could just call the `add` twice.

Reply:
We can fix in other places as well.

Reply:
Then won't the outputs look like `12.0 | f | 13.0 | f` instead of `12.0f | 13.0f` and `" | Foo | " | " | Bar | "` instead of `"Foo" | "Bar"`?

Btw I think I need to change `new StringJoiner(" | ");` to `new StringJoiner("|");` as done in other places.

Reply:
+1